src/__tests__/baseFieldLocalizations.int.test.ts

```diff
@@ -0,0 +1,322 @@
+					description: 'Date of Birth',
+				})
+				.expect(409);
+			expect(result.body).toMatchObject({
+				name: 'DatabaseError',
+				details: [
+					{
+						code: PostgresErrorCode.UNIQUE_VIOLATION,
+					},
+				],
+			});
+		});
+
+		it('returns 422 conflict when a base field is referenced that does not exist', async () => {
+			const result = await agent
+				.post('/baseFieldLocalizations')
+				.type('application/json')
+				.set(adminUserAuthHeader)
+				.send({
+					baseFieldId: 1,
+					language: 'en',
+					label: 'First Name',
+				})
+				.expect(422);
+			expect(result.body).toMatchObject({
+				name: 'DatabaseError',
+				details: [
+					{
+						code: PostgresErrorCode.FOREIGN_KEY_VIOLATION,
+					},
+				],
+			});
+		});
+	});
+
+	describe('PUT /baseFieldLocalizations', () => {
+		it('requires authentication', async () => {
+			await agent.put('/baseFieldLocalizations').expect(401);
+		});
+
+		it('requires administrator role', async () => {
+			await agent.put('/baseFieldLocalizations').set(authHeader).expect(401);
+		});
+
+		it('updates the specified base field localization', async () => {
+			// Not using the helper here because observing a change in values is explicitly
+			// the point of the test, so having full explicit control of the original value
+			// seems important.  Some day when we add better test tooling we can have it all.
+			// That day is not yet.
+			const baseField = await createBaseField({
+				shortCode: 'summary',
+				dataType: BaseFieldDataType.STRING,
+				scope: BaseFieldScope.PROPOSAL,
+			});
+			await createBaseFieldLocalization({
+				baseFieldId: baseField.id,
+				language: 'en',
+				label: 'Summary',
+				description: 'Summary Description',
+			});
+			await agent
+				.put('/baseFieldLocalizations')
+				.type('application/json')
+				.set(adminUserAuthHeader)
+				.send({
+					baseFieldId: baseField.id,
+					language: 'en',
+					label: 'Soomary',
+					description: 'Soooomary Description',
+				})
+				.expect(200);
+			const baseFieldLocalizations = await loadBaseFieldLocalizations();
+			expect(baseFieldLocalizations[0]).toMatchObject({
+				baseFieldId: baseField.id,
+				language: 'en',
+				label: 'Soomary',
+				description: 'Soooomary Description',
+				createdAt: expectTimestamp,
+			});
+		});
+
+		it('returns the updated base field localization', async () => {
+			const baseFieldId = await createTestBaseField();
+			const result = await agent
+				.put('/baseFieldLocalizations')
+				.type('application/json')
+				.set(adminUserAuthHeader)
+				.send({
+					baseFieldId,
+					language: 'en',
+					label: 'Soomary',
+					description: 'Soooomary Description',
+				})
+				.expect(200);
+			expect(result.body).toMatchObject({
+				baseFieldId,
+				language: 'en',
+				label: 'Soomary',
+				description: 'Soooomary Description',
+				createdAt: expectTimestamp,
+			});
+		});
+
+		it('returns 400 bad request when no language is sent', async () => {
```

Comment:
Should we add a test for when an invalid language code is sent (and should we add support for ensuring that a language is valid...)

A potential lead for implementation of this concept: https://www.npmjs.com/package/i18n-iso-countries

Reply:
I agree with the need for this test! I think we want to validate specifically against a list of language codes rather than countries, though.

We'd previously talked about using the `en-US` style tags, which it turns out are called [IETF language tags](https://en.wikipedia.org/wiki/IETF_language_tag). There is a package that has the current list and functions to work with it: [language-tags](https://www.npmjs.com/package/language-tags).

Reply:
@jasonaowen I could be wrong but it seems like this package doesn't support ESModule syntax? I'm not really sure what that means, but it won't let me import anything from it in our application, i'd have to use the CommonJS 'require'. Likely this is easily resolvable but wanted to defer to your judgement here

Reply:
@hminsky2002 what are you trying to do? I was able to call it like this:

```ts
import * as tags from 'language-tags';
logger.info(tags.default('en-US'), 'Testing language-tags');
```

and get a log message like
```
[2024-08-02T23:03:53.688Z] INFO (219438): Testing language-tags
    source: "pdc/service/src/index.ts"
    data: {
      "tag": "en-us"
    }
```

Reply:
@jasonaowen oh ok weird! Must be a bug with my IDE then:
<img width="656" alt="Screenshot 2024-08-02 at 7 11 34 PM" src="https://github.com/user-attachments/assets/9cb4d0d9-795c-411a-a069-807ed0873410">
maybe my intellisense is just bugged? That seems like a me problem and not an issue as far as the codebase is concerned, as long as it compiled

Reply:
Oh, as the error message suggests, you also need to install `@types/language-tags`! That's often the solution for using packages without built-in type declarations.

Reply:
Ahhh, that makes sense. I had assumed that such a package did not actually exist, how foolish

Reply:
@hminsky2002 the main reason TypeScript is actually usable is that there is a huge community of people who have written type definitions for various node packages that don't include type definitions (check it out at https://github.com/DefinitelyTyped/DefinitelyTyped/)

This is a huge huge repo and it exports @types/...

And note that you can contribute type definitions there too if you ever come across a library lacking them!